component/src/main/java/io/siddhi/extension/io/file/function/FileCopyExtension.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.siddhi.extension.io.file.function;
+
+import io.siddhi.annotation.Example;
+import io.siddhi.annotation.Extension;
+import io.siddhi.annotation.Parameter;
+import io.siddhi.annotation.ReturnAttribute;
+import io.siddhi.annotation.util.DataType;
+import io.siddhi.core.config.SiddhiQueryContext;
+import io.siddhi.core.exception.SiddhiAppRuntimeException;
+import io.siddhi.core.executor.ExpressionExecutor;
+import io.siddhi.core.executor.function.FunctionExecutor;
+import io.siddhi.core.util.config.ConfigReader;
+import io.siddhi.core.util.snapshot.state.State;
+import io.siddhi.core.util.snapshot.state.StateFactory;
+import io.siddhi.extension.io.file.util.Constants;
+import io.siddhi.extension.io.file.util.VFSClientConnectorCallback;
+import io.siddhi.query.api.definition.Attribute;
+import io.siddhi.query.api.exception.SiddhiAppValidationException;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.log4j.Logger;
+import org.wso2.carbon.messaging.BinaryCarbonMessage;
+import org.wso2.carbon.messaging.exceptions.ClientConnectorException;
+import org.wso2.transport.file.connector.sender.VFSClientConnector;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.siddhi.extension.io.file.util.Constants.WAIT_TILL_DONE;
+import static io.siddhi.query.api.definition.Attribute.Type.BOOL;
+import static io.siddhi.query.api.definition.Attribute.Type.STRING;
+
+/**
+ * This extension can be used to copy files from a particular source to a destination.
+ */
+@Extension(
+        name = "copy",
+        namespace = "file",
+        description = "This function performs copying file from a particular source to a destination.\n",
```

Comment:
dont use the word "source" say "from one directory to another"

Reply:
@suhothayan fixed with https://github.com/siddhi-io/siddhi-io-file/pull/72/commits/4eaba8254220943cd983761f19554006840f5d17